### PR TITLE
ThreadRunner: use std::thread

### PR DESCRIPTION
**Pull request by whyman**
More work towards tidying up the statics.

With this patch I can run load testing tool successfully against a transcoded image resource path without it crashing in various places.

### CODE REVIEW

File: src/util/thread_runner.h [hunk 1]
```diff
@@ -31,36 +31,31 @@
 #include <mutex>
 #include <thread>
 
-#include <pthread.h>
-
-#include "config/config.h"
 #include "thread_executor.h"
 
 using ThreadProc = void* (*)(void* target);
```
Comment (neheb) [inline src/util/thread_runner.h:36]: Can this be replaced with std::function?

File: src/util/thread_runner.h [hunk 2]
```diff
@@ -164,56 +172,14 @@ class ThreadRunner : public ThreadExecutor {
...
-    }
-
-    /// \brief start the thread
-    void startThread() override
-    {
-#if !defined(SOLARIS) && !defined(__HAIKU__)
-        // default scoping on Solaroid systems is in fact PTHREAD_SCOPE_SYSTEM
-        // plus, setting PTHREAD_EXPLICIT_SCHED requires elevated privileges
-        // while Haiku doesn't implement pthread_attr_setinheritsched yet
-        if (config->getBoolOption(CFG_THREAD_SCOPE_SYSTEM)) {
```
Comment (KarlStraussberger) [inline src/util/thread_runner.h:184]: You can also drop the option from config

**Review by KarlStraussberger — COMMENTED** [review]
Seems to make multi-threading a lot clearer.

**Comment by neheb** [comment]
```
/Users/runner/work/gerbera/gerbera/src/content/onlineservice/atrailers_service.cc:43:15: error: member access into incomplete type 'std::shared_ptr<Config>::element_type' (aka 'Config')
    if (config->getOption(CFG_ONLINE_CONTENT_ATRAILERS_RESOLUTION) == "640")
              ^
/Users/runner/work/gerbera/gerbera/src/context.h:32:7: note: forward declaration of 'Config'
class Config;
```

File: src/util/thread_runner.h (hunk 2 above)
Comment (whyman) [reply src/util/thread_runner.h:184]: Good spot!

**Comment by whyman** [comment]
Still working on this, trying to debug some fun crashes in buffered IO handling.

**Comment by whyman** [comment]
Closing for now, working on new approach.

**Pull request closed without merging**